public/index.php

```diff
@@ -54,7 +53,13 @@
     Request::setTrustedHosts(explode(',', $trustedHosts));
 }
 
-$kernel = new Kernel($env, $debug, SuluKernel::CONTEXT_WEBSITE);
+$suluContext = SuluKernel::CONTEXT_WEBSITE;
+
+if (preg_match('/^\/admin(\/|$)/', $_SERVER['REQUEST_URI'])) {
```

Comment:
Maybe we can add a env variable for this slug `/admin`.
When it's not set, it's default and we use `/admin`.

Reply:
https://stackoverflow.com/questions/6433492/preg-match-vs-strpos-for-match-finding

Reply:
This file ist editable by the project. /admin is also hardcoded in security_admin.yaml and routes_admin.yaml.

Reply:
@chirimoya was also my first intention but would block website urls like `/administration`

Reply:
What if you use a combination of `strpos` and `strlen`? Would probably require a benchmark first, to see which one is faster. But definitely worth it, since it is executed on every request :slightly_smiling_face:

Reply:
@danrot @chirimoya in case of the website where we should be optimize the performance for the preg_match seems to be faster (the equal and strpos would be faster for the admin):

```php
time php -r "\$server=['REQUEST_URI' => '/hello']; \$i=0;while(++\$i < 10000000) {if(preg_match('/^\/admin(\/|$)/', \$server['REQUEST_URI'])) { }}"
php -r   7,90s user 0,01s system 99% cpu 7,912 total
```

```php
time php -r "\$server=['REQUEST_URI' => '/hello']; \$i=0;while(++\$i < 10000000) {if('/admin' === \$server['REQUEST_URI'] || 0 === strpos('/admin/', \$server['REQUEST_URI'])) {}}"
php -r   8,56s user 0,01s system 99% cpu 8,574 total
```